app/routes/setup-page.tsx

```diff
@@ -125,3 +122,57 @@ export default function SetupPage() {
     </>
   );
 }
```

Comment:
I've noticed that you've stuck with handling state in the client-side React component not using any Remix handlers or anything (e.g. ClientAction). That was one of the things that I couldn't get my head around (hence my working code that I pushed just used pure React). I guess we just roll with straightup React for now?

Reply:
(Next commit) Oh, cool. I see that you've used a ref rather than useState. Makes sense. I think I had come across something suggesting that.